### DISCUSSION TITLE
Group "Upcoming" Tasks into sections

### DISCUSSION THREAD

**Discussion by JarWarren** [twentyhq/twenty#6006] [Ideas]
## Scope & Context

Ability to see Tasks grouped into sections, based on the day they're scheduled. June 27, June 28, June 30, etc. Perhaps even showing the day of the week. "Monday, June 30th".

Maybe this could be the default behavior, or perhaps an opt-in setting.

## Current behavior

Right now all future Tasks are grouped into a single "Upcoming" section, not sorted by date.

**Comment by greptile-apps** [comment]
To group 'Upcoming' tasks into sections based on the day they're scheduled, follow these steps:

1. **Modify `useTasks` Hook**: Update the `useTasks` hook to categorize tasks by date.
   - File: `src/hooks/useTasks.js`
   - Add logic to group tasks by their scheduled date.

2. **Update `Tasks` Component**: Adjust the `Tasks` component to render tasks in the new grouped structure.
   - File: `src/components/Tasks.js`
   - Implement rendering logic to display tasks grouped by date sections.

3. **Adjust `TaskRow` Component**: Modify the `TaskRow` component if necessary to support the new grouped structure.
   - File: `src/components/TaskRow.js`
   - Ensure it can handle the new date-grouped format.

Example code snippets and detailed implementation should be added directly in the respective files.



### References

**Comment by Bonapara** [comment]
Hi @JarWarren Great idea! I think we will try to move the task object to a regular object view so it can benefits from the filters of other objects
